scripts/table_updates/update_data_table.py

```diff
@@ -423,79 +454,258 @@ def update_redact_table(
+        cohort (str, optional): The cohort name. Defaults to "".
+
+    Returns:
+        Tuple[str, pandas.DataFrame]: The synapse ID for BPC table to be modified and the updated table as dataframe
+    """
+    # check the validity of bpc_column_list
+    valid_col = column_mapping_table.loc[
+        column_mapping_table["prissmm_form"] == form,
+    ].prissmm_element.tolist()
+    assert all(
+        item in valid_col for item in bpc_column_list
+    ), f"Invalid bpc_column_list. Column names should be matching {valid_col}."
+
+    logger.info(f"Update {bpc_column_list} in {form}")
+    # load bpc table
+    cpt_table_id = master_table.loc[master_table["form"] == form, "id"].values[0]
+
+    if cohort:
+        condition = f"cohort = '{cohort}'"
+        cpt_dat = utilities.download_synapse_table(
+            syn, cpt_table_id, condition=condition
+        )
+    else:
+        cpt_dat = utilities.download_synapse_table(syn, cpt_table_id)
+    cpt_dat.index = cpt_dat.index.map(str)
+    cpt_dat["index"] = cpt_dat.index
+    # subset main_genie_table based on bpc_column_list
+    main_genie_column_list = [
+        ", ".join(
+            column_mapping_table.loc[
+                column_mapping_table["prissmm_element"] == col,
+            ].genie_element
+        )
+        for col in bpc_column_list
+    ]
+
+    if form == "patient_characteristics":
+        main_genie_table = main_genie_table[main_genie_column_list + ["PATIENT_ID"]]
+        # check if there are genie_patient_ids in bpc table but not in main genie table
+        check_if_all_join_keys_available(
+            left_table=cpt_dat,
+            right_table=main_genie_table,
+            left_on="genie_patient_id",
+            right_on="PATIENT_ID",
+            logger=logger,
+        )
+        cpt_seq_dat = cpt_dat.merge(
+            main_genie_table,
+            how="left",
+            left_on="genie_patient_id",
+            right_on="PATIENT_ID",
+        )
+        utilities.update_tier1a_data_replacement_mapping_table(
+            syn, cpt_seq_dat, form, config
+        )
+    else:
+        # for cancer_panel_test table
+        main_genie_table = main_genie_table[main_genie_column_list + ["SAMPLE_ID"]]
+        # check if there are cpt_genie_sample_ids in bpc table but not in main genie table
+        check_if_all_join_keys_available(
+            left_table=cpt_dat,
+            right_table=main_genie_table,
+            left_on="cpt_genie_sample_id",
+            right_on="SAMPLE_ID",
+            logger=logger,
+        )
+        cpt_seq_dat = cpt_dat.merge(
+            main_genie_table,
+            how="left",
+            left_on="cpt_genie_sample_id",
+            right_on="SAMPLE_ID",
+        )
+        utilities.update_tier1a_data_replacement_mapping_table(
+            syn, cpt_seq_dat, form, config
+        )
+    # reformat the columns
+    cpt_seq_dat.index = cpt_seq_dat["index"]
```

Comment:
Since `cpt_seq_dat` goes through a bunch of formatting here and I don't see tests for `update_tier1a` function, might be worth putting these in a separate function and testing that. I don't know how much this section of code changed from previous

Reply:
Oh this is because I forgot to push the test scripts for update_tier1a and will push it with other changes.